presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -777,8 +778,8 @@ private static InnerJoinPushDownResult processInnerJoin(Expression inheritedPred
             joinConjuncts.addAll(filter(extractConjuncts(joinPredicate), not(DeterminismEvaluator::isDeterministic)));
             joinPredicate = filterDeterministicConjuncts(joinPredicate);
 
-            leftEffectivePredicate = filterDeterministicConjuncts(leftEffectivePredicate);
-            rightEffectivePredicate = filterDeterministicConjuncts(rightEffectivePredicate);
+            leftEffectivePredicate = stripDynamicFilters(filterDeterministicConjuncts(leftEffectivePredicate));
```

Comment:
dynamic filters might come from `JoinNode#getFilter` itself and not only from left or right side.
Also not all dynamic filters in build side are unwanted. For example subtree join build side might be probe side to a join downstream.

I was more thinking that `RemoveUnsupportedDynamicFilters` should remove dynamic filters from join build side as it can pass context top-down which dynamic filters are allowed

Reply:
I've reverted this change and modified RemoveUnsupportedDynamicFilters to remove dynamic filters from join build side.